src/NATS.Client/JetStream/JetStream.cs

```diff
@@ -88,6 +89,24 @@ private PublishAck PublishSyncInternal(string subject, byte[] data, MsgHeader hd
             return ProcessPublishResponse(Conn.Request(msg), options);
         }
 
+        private Task<PublishAck> PublishAsyncInternal(string subject, byte[] data, MsgHeader hdr, PublishOptions options)
+        {
+            MsgHeader merged = MergePublishOptions(hdr, options);
+            Msg msg = new Msg(subject, null, merged, data);
+
+            if (JetStreamOptions.IsPublishNoAck)
+            {
+                Conn.Publish(msg);
+                return null;
+            }
+
+            Duration timeout = options == null ? JetStreamOptions.RequestTimeout : options.StreamTimeout;
+
+            return Conn.RequestAsync(msg, timeout.Millis)
+                .ContinueWith(async antecedent => ProcessPublishResponse(antecedent.Result, options), CancellationToken.None)
```

Comment:
Drop the `async` if you are not using `await`:
```suggestion
                .ContinueWith(antecedent => ProcessPublishResponse(antecedent.Result, options), CancellationToken.None)
```

Reply:
On second thought, is there a reason that `PublishAsyncInternal` cannot be declared `async` and you `await Conn.RequestAsync(...)`?

Reply:
Conn.RequestAsync already returns a future. It hides the async under the covers. I understood that antecedent just tacks itself on to the task.

Reply:
In general you don't really want to use ContinueWith if it can be avoided. Lots of little caveats here and there as to how the task is attached, cancellation, exceptions; best to use `async` + `await` if you're not just immediately chaining a result. It can also get you into trouble if at any point you mix sync-over-async and async code.

Reply:
Adding an issue so we don't lose track of this. https://github.com/nats-io/nats.net/issues/475